benchmarks/pytorch_benchmark_rmsnorm_backward.py

```diff
@@ -0,0 +1,377 @@
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import torch
+import torch.nn as nn
+
+from quack.rmsnorm import QuackRMSNorm
+from tabulate import tabulate
+
+
+class RMSNorm(nn.Module):
+    def __init__(self, dim, eps=1e-5):
+        super().__init__()
+        self.eps = eps
+        self.scale = nn.Parameter(torch.ones(dim))
+
+    def forward(self, x):
+        norm_x = x.norm(2, dim=-1, keepdim=True)
+        rms_x = norm_x * torch.rsqrt(torch.tensor(x.shape[-1], dtype=x.dtype))
+        x_normed = x / (rms_x + self.eps)
+        return self.scale * x_normed
+
+
+def benchmark_backward_implementation(
+    implementation_name,
+    model,
+    input_data,
+    num_iterations=100,
+    warmup_iterations=10,
+):
+    """Benchmark a specific implementation's backward pass and return timing results."""
+    # Create gradient tensor of the same shape as the output
+    grad_output = torch.randn_like(input_data)
+
+    # Warmup
+    for _ in range(warmup_iterations):
+        input_data_clone = input_data.clone().requires_grad_(True)
+        output = model(input_data_clone)
+        output.backward(grad_output)
+        torch.cuda.synchronize()
+
+    # Benchmark
+    start_event = torch.cuda.Event(enable_timing=True)
+    end_event = torch.cuda.Event(enable_timing=True)
+
+    start_event.record()
+    for _ in range(num_iterations):
+        input_data_clone = input_data.clone().requires_grad_(True)
```

Comment:
this timing would include the cloning time, not just the backward time?

Reply:
fixed!